src/rdkafka.h

```diff
@@ -1385,6 +1496,14 @@ void rd_kafka_conf_set_dr_cb(rd_kafka_conf_t *conf,
  *
  * An application must call rd_kafka_poll() at regular intervals to
  * serve queued delivery report callbacks.
+ *
+ * The broker-assigned offset can be retrieved with \c rkmessage->offset
+ * ((if `producer.offset.report` is true), and the timestamp can be
+ * retrieved using rd_kafka_message_timestamp().
+ *
+ * @remark The Idempotent Producer may return invalid timestamp
+ *         and offset for retried messages that were previously
+ *         successfully delivered but not properly acknowledged.
```

Comment:
after reading this i'm wondering:
1. can there potentially be more than one DR for a given message (assume no)?
2. could the DR returned for a successfully delivered message have invalid properties - timestamp and offset?
3. could the DR for a successfully produced message have an error code != `NoError`?

if yes to any of the above, doesn't seem ideal.

Reply:
1. No
 2. Yes, if the message was retried and got the ERR__DUPLICATE error back from the broker, which indicates that the message was persisted on the first produce but the ack was lost. But there is nothing we can do about it but document the behaviour.
 3. No